src/preamble.js

```diff
@@ -170,6 +177,7 @@ function ccall (ident, returnType, argTypes, args, opts) {
 #endif
 #endif
   if (returnType === 'string') ret = Pointer_stringify(ret);
+  if (returnType === 'bool') ret = Boolean(ret);
```

Comment:
this part looks correct, yeah, without it the return value would be a number and not a boolean.

please change it to `boolean`, though, so it matches the others (which are JS types, the same as `typeof` returns), and update the docs to mention it (https://kripken.github.io/emscripten-site/docs/api_reference/preamble.js.html#ccall , in the repo it's under `site/`).